### PR TITLE
use course timezones in builder

### PR DESCRIPTION
**Pull request by jivey**
- Update the tasking to use the course timezone when making date calculations
- Use momentInZone in the DateTimeInput when rendering values (if course was passed)
- Fix a bug that prevented opens at from being changed if due at wasn't changed (changed `!due_at` check to `due_at`, now matches `onDueChange`.)
- ~~Recalculate course bounding when changing the course time_zone. (This prevents clicking on the start day of a course, and by changing the time_zone to an earlier one, causing the opens_at to be before the first start date. For example, `July 1st 12:01` can become `June 30 11:01PM`~~ Removed, as the entire timezone modal is being removed from the builder
- Use course timezone anywhere we display times on the teacher side
- Use browser timezone on the student side (this already happens because a bare moment call will use the browser, but you have to use `tz.guess()` to actually get the timezone to display when calling `format` with `'z'`.
- Make some small design adjustments to make the wider time displays fit

image

image

image

image

image

### CODE REVIEW

File: tutor/src/screens/assignment-edit/tasking.js [hunk 1]
```diff
@@ -242,6 +219,7 @@ class Tasking extends React.Component {
               name={`tasking_plans[${index}].closes_at`}
               onChange={this.onClosesChange}
               disabledDate={this.course.isInvalidAssignmentDate}
+              course={this.course}
```
Comment (camliu89) [inline tutor/src/screens/assignment-edit/tasking.js:222]: Maybe just pass the function `momentInZone` instead of the whole course?

Comment (jivey) [reply tutor/src/screens/assignment-edit/tasking.js:222]: That's a great idea, I'm going to update that and fix up a different issue I found.

File: tutor/specs/screens/student-dashboard/__snapshots__/dashboard.spec.jsx.snap [hunk 2]
```diff
@@ -513,7 +513,7 @@ exports[`Student Dashboard displays as loading 1`] = `
                       className="sc-fznyAO c2"
                     >
                       <time>
-                        We Oct 21, 7:00am
```
Comment (jivey) [inline tutor/specs/screens/student-dashboard/__snapshots__/dashboard.spec.jsx.snap:516]: Just realized these snapshots might change depending on where they are run 🤔

**Comment by jivey** [comment]
Snapshot failures are probably from the CI needing to have the time configuration changed? I reran all the snapshots in CT so other devs at least should see them passing.

```
- Wed, Oct 21, 6:00am CST
+ Wed, Oct 21, 6:00pm +06
```

File: tutor/src/screens/assignment-review/details.js [hunk 3]
```diff
@@ -240,16 +236,25 @@ const TaskingDates = observer(({ tasking, title }) => {
...
         </div>
         {!tasking.plan.isEvent && (
           <div>
             <dt>Close date</dt>
-            <dd>{format(tasking.closesAtMoment)}</dd>
+            <dd>
+              <div>{tasking.closesAtMoment.format(dateFormat)}</div>
+              <div>{tasking.closesAtMoment.format(timeFormat)}</div>
+            </dd>
           </div>)}
```
Comment (nathanstitt) [inline tutor/src/screens/assignment-review/details.js:258]: I'm probably missing something?  Why are each of the dates listed twice?

Comment (jivey) [reply tutor/src/screens/assignment-review/details.js:258]: It breaks the date and time into separate divs so the line break is nicer than wrapping

Comment (nathanstitt) [reply tutor/src/screens/assignment-review/details.js:258]: aha! I completely missed the date/time difference there

File: tutor/src/screens/student-dashboard/event-row.jsx [hunk 4]
```diff
@@ -105,7 +109,7 @@ class EventRow extends React.Component {
             <span>{event.title}</span>
           </TitleCell>
           <DueCell>
-            <EventTime event={event} />
+            <EventTime event={event} course={course} />
             {event.is_extended && <StyledEIcon />}
```
Comment (nathanstitt) [inline tutor/src/screens/student-dashboard/event-row.jsx:113]: course doesn't seem to be used in EventTime

Comment (jivey) [reply tutor/src/screens/student-dashboard/event-row.jsx:113]: Oops good catch, forgot I changed course there. Weirdly, lint didn't tell me!

**Pull request merged**